olio/service/resources/version_resource_test.go

```diff
@@ -0,0 +1,61 @@
+package resources
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+type Extractor struct {
+}
+
+func NewExtractor() Extractor {
+	return Extractor{}
+}
+
+func (e Extractor) GetVersion() string {
+	return "0.0.11"
+}
+
+func (e Extractor) GetAppName() string {
+	return "cx-messaging"
+}
+
+func TestVersion(t *testing.T) {
+	versionExtractor := NewExtractor()
+
+	tt := []struct {
+		name string
+	}{
+		{name: "No extractor"},
+		{name: "With extractor"},
```

Comment:
Should contain the expected output as well, and test below refactored so the repeated aspects of both cases (all except one line 'versionResource.AddVersionExtractor(versionExtractor)') are DRY-ed up